src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs

```diff
@@ -93,8 +93,9 @@ public override bool RunTask ()
 				.Concat (MonoAndroidHelper.GetFrameworkAssembliesToTreatAsUserAssemblies (ResolvedAssemblies))
 				.ToList ();
 			var mainFileName = Path.GetFileName (MainAssembly);
+			var mainAssembly = new List<ITaskItem> () { new TaskItem (mainFileName) };
 			Func<string,string,bool> fileNameEq = (a,b) => a.Equals (b, StringComparison.OrdinalIgnoreCase);
-			assemblies = assemblies.Where (a => fileNameEq (a.ItemSpec, mainFileName)).Concat (assemblies.Where (a => !fileNameEq (a.ItemSpec, mainFileName))).ToList ();
+			assemblies = mainAssembly.Concat (assemblies.Where (a => !fileNameEq (Path.GetFileName (a.ItemSpec), mainFileName))).ToList ();
```

Comment:
Since we're changing this, should we take a moment and remove the System.Linq?

Maybe it could just be two `foreach` loops, that write as they go? Print all the matches for `mainFileName`, then the rest in a second pass?

Reply:
Good call